addons/account/migrations/9.0.1.1/post-migration.py

```diff
@@ -292,17 +292,55 @@ def account_internal_type(cr):
             })
 
 
-@openupgrade.migrate()
-def migrate(cr, version):
+def account_partial_reconcile(env):
+    # disable all workflow steps
+    set_workflow_org = models.BaseModel.step_workflow
+    models.BaseModel.step_workflow = lambda *args, **kwargs: None
+    cr = env.cr
+    move_line_ids = {}
+    cr.execute("SELECT reconcile_id, id FROM account_move_line WHERE "
+               "reconcile_id IS NOT null")
+    for rec_id, move_line_id in cr.fetchall():
+        if rec_id not in move_line_ids.keys():
+            move_line_ids[rec_id] = [move_line_id]
+        else:
+            move_line_ids[rec_id] += [move_line_id]
+    move_lines_1 = env['account.move.line']
+    for rec_id in move_line_ids.keys():
+        move_lines_1 = env['account.move.line'].browse(
+            [i for i in move_line_ids[rec_id]])
+        move_lines_1.auto_reconcile_lines()
+    move_line_ids = {}
+    cr.execute("SELECT reconcile_partial_id, id FROM account_move_line WHERE "
+               "reconcile_partial_id IS NOT null")
+    for rec_id, move_line_id in cr.fetchall():
+        if rec_id not in move_line_ids.keys():
+            move_line_ids[rec_id] = [move_line_id]
+        else:
+            move_line_ids[rec_id] += [move_line_id]
+    move_lines_2 = env['account.move.line']
+    for rec_id in move_line_ids.keys():
+        move_lines_2 = env['account.move.line'].browse(
+            [i for i in move_line_ids[rec_id]])
+        move_lines_2.auto_reconcile_lines()
+    for field in ['amount_residual', 'amount_residual_currency', 'reconciled']:
+        env.add_todo(env['account.move.line']._fields[field],
+                     move_lines_1 + move_lines_2)
```

Comment:
Scoping is messed up. move_lines_1 & 2 will only contain the move lines from the last reconciliation in each loops, which is not what you want. Fixed here: https://github.com/OCA/OpenUpgrade/pull/615